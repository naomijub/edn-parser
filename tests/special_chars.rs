#![allow(missing_docs)]

use edn_parser::*;
use insta::assert_snapshot;

#[test]
fn newline() {
    assert_snapshot!(edn_parse("\\newline").unwrap().cst);
}

#[test]
fn tab() {
    assert_snapshot!(edn_parse("\\tab").unwrap().cst);
}

#[test]
fn return_char() {
    assert_snapshot!(edn_parse("\\return").unwrap().cst);
}

#[test]
fn space() {
    assert_snapshot!(edn_parse("\\space").unwrap().cst);
}

#[test]
fn unicode() {
    assert_snapshot!(edn_parse("\\u0000").unwrap().cst);
}

#[test]
fn non_unicode_short_err() {
    let error = edn_parse("\\u000").unwrap_err();
    let error_json = serde_json::to_string_pretty(&error).unwrap();
    assert_snapshot!(error_json);
}

#[test]
fn non_unicode_long_err() {
    let error = edn_parse("\\u00000").unwrap_err();
    let error_json = serde_json::to_string_pretty(&error).unwrap();
    assert_snapshot!(error_json);
}

#[test]
fn random_err() {
    let error = edn_parse("\\random").unwrap_err();
    let error_json = serde_json::to_string_pretty(&error).unwrap();
    assert_snapshot!(error_json);
}
