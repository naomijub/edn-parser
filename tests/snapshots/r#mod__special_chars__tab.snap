---
source: tests/special_chars.rs
expression: "edn_parse(\"\\\\tab\").unwrap().cst"
---
edn [0..4]
    composed [0..4]
        char [0..4]
            CharEscape "\\" [0..1]
            Chars "tab" [1..4]
